src/components/UI/atoms/Button/Button.tsx

```diff
@@ -0,0 +1,15 @@
+import { PropsWithChildren, ButtonHTMLAttributes,  useMemo } from "react";
+import  "./styles.module.css";
+
+type ButtonNativeTypes = ButtonHTMLAttributes<HTMLButtonElement>;
+
+const Button = ({ children, className, ...props }: PropsWithChildren<ButtonNativeTypes>): JSX.Element => {
+  const styles = useMemo<string>(() => className ?? "button", [className]);
```

Comment:
This could be a bit controversial but I believe there is no need to be using `useMemo` here, it should be used only for optimization purposes, meaning that you already measured there is some value calculation affecting the performance of your application. You could revisit [react docs](https://react.dev/reference/react/useMemo#skipping-expensive-recalculations) and see if it makes sense.